### PR TITLE
Display heading on GPS Tab

### PR DESCRIPTION
**Pull request by atomgomba**
Use data from magnetometer (when enabled) to display heading in degrees and rotate the map accordingly. Can be used to verify correct mag alignment or for testing in general. This PR also adds Virtual FC support for GPS Tab so when connected to VFC pre-defined sample data and thus the map as well will show up.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/tabs/map.js
Comment (haslinghuis) [inline src/js/tabs/map.js:103]: Why remove string interpolation?

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Good idea !!!

File: src/js/VirtualFC.js
Comment (atomgomba) [reply src/js/VirtualFC.js:247]: This is interesting, because I copied the object from memory to clipboard using `copy(FC.GPS_DATA)` in the console. I have to double check this, I think the typo is fundamentally somewhere else...

File: src/js/tabs/map.js
Comment (atomgomba) [reply src/js/tabs/map.js:103]: So that we let the browser format the error object as it wishes and not force it to be a string.

Comment (atomgomba) [reply src/js/tabs/map.js:103]: Plus, it won't really show since the map is running in its own webview with its own console. You have to have the console of the inner webview open to see the message anyway. It's bad, but I didn't take the time to look deeper in the issue

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Oops, duplicate found:

image

**Comment by atomgomba** [comment]
> Oops, duplicate found:
> 
> image

Oh, you have find the typo! I believe the original typo was in `fc.js` but it was just a default value which didn't cause a problem, since on every update in `MSPHelper.js` the name was correct and this field isn't used in BFC at all! Nice catch!

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
* is everything resolved here?
* can @atomgomba  add an photo/screenshot of what it does in the description?

**Review by nerdCopter — APPROVED** [review]
nice! approving. watched demo video

**Review by blckmn — APPROVED** [review]

**Pull request merged**
